multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/FilesApiServiceImpl.java

```diff
@@ -138,4 +154,55 @@ private FileMetadata parseFileEntry(FileEntry fileEntry) {
                                     .namespace(fileEntry.getNamespace())
                                     .build();
     }
+
+    private FileEntry uploadFileFromUrl(String spaceGuid, String namespace, String mtarEndpoint) throws FileStorageException, IOException {
+        if (!UriUtil.isValidUrl(mtarEndpoint)) {
+            throw new SLException("Invalid URL");
+        }
+        String decodedUrl = URLDecoder.decode(mtarEndpoint, StandardCharsets.UTF_8);
+        WebClient client = new RestUtil().createWebClient(false);
```

Comment:
Not sure if this is a good idea. `RestUtil` is more of an internal class for the client facade. It'd be nice to have the option of changing it whenever we want, without having to touch the deploy-service codebase.

Reply:
Agreed. 
Changed to `WebClientFactory::getWebClient()` which is in the deploy-service codebase